### PR TITLE
Pre-cache MNI2009 and MNIInfant templates

### LINKED ISSUE #4: Use Templateflow to download necessary templates
At minimum, QSIRecon, requires the res-1 MNI152NLin2009cAsym and MNIInfant (cohort-2) T1w templates, along with the associated brain masks.

I think this would replace https://github.com/PennLINC/qsirecon_build/blob/7b466befcaf1ebac2a9a14080993566affe70c0c/docker/scripts/get_templates.sh.

Related to https://github.com/PennLINC/qsirecon/pull/49.

### PR DESCRIPTION
**Pull request by tsalo**
Closes #4.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -204,23 +204,27 @@ ENV \
 
 WORKDIR /root/
 
-# Precaching atlases + AtlasPack
+# Precaching templates
+COPY scripts/fetch_templates.py fetch_templates.py
+RUN ${CONDA_PYTHON} -m pip install --no-cache-dir --upgrade templateflow && \
```
Comment (mattcieslak) [inline Dockerfile]: I think templateflow is already installed in the conda python [here](https://github.com/PennLINC/qsiprep_build/blob/main/qsiprep_env.yml#L38). Was it not showing up as available in the build?

Comment (tsalo) [reply Dockerfile]: Sorry, I just copied over this section from the fMRIPrep Dockerfile. I'll drop the install.

Comment (mattcieslak) [reply Dockerfile]: I'm downloading a copy of qsirecon_build just to make sure templateflow is there

Comment (mattcieslak) [reply Dockerfile]: it's in there

Comment (tsalo) [reply Dockerfile]: Sweet! I'll resolve this then. Any other concerns about the PR?

**Review by mattcieslak — APPROVED** [review]

**Pull request merged**